apps/android/app/src/main/java/chat/simplex/app/views/chat/SendMsgView.kt

```diff
@@ -35,185 +36,56 @@ import androidx.core.view.inputmethod.InputConnectionCompat
-            if (cs.preview !is ComposePreview.VoicePreview && filePath.value != null) {
-              // Pressed on X icon in preview
-              cleanUp(true)
-            }
-          }
-          val interactionSource = interactionSourceWithTapDetection(
-            // It's just a key for triggering dropping a state in the compose function. Without it
-            // nothing will react on changed params like needToAllowVoiceToContact or allowedVoiceByPrefs
-            needToAllowVoiceToContact.toString() + allowedVoiceByPrefs.toString(),
-            onPress = {
-              if (filePath.value == null) startStopRecording()
-            },
-            onClick = {
-              // Voice not allowed or not granted voice record permission for the app
-              if (!allowedVoiceByPrefs || !permissionsState.allPermissionsGranted) return@interactionSourceWithTapDetection
-              if (!recordingInProgress.value && filePath.value != null) {
-                sendMessage()
-                cleanUp(false)
-              } else if (stopRecOnNextClick) {
-                stopRecordingAndAddAudio()
-                stopRecOnNextClick = false
-              } else {
-                // tapped and didn't hold a finger
-                stopRecOnNextClick = true
-              }
-            },
-            onCancel = startStopRecording,
-            onRelease = startStopRecording
-          )
-          val sendButtonModifier = if (recordingTimeRange.last != 0L)
-            Modifier.clip(CircleShape).background(color)
-          else
-            Modifier
-          IconButton({}, Modifier.size(36.dp), enabled = !cs.inProgress, interactionSource = interactionSource) {
-            Icon(
-              when {
-                recordingTimeRange.last != 0L -> Icons.Outlined.ArrowUpward
-                stopRecOnNextClick -> Icons.Filled.Stop
-                allowedVoiceByPrefs -> Icons.Filled.KeyboardVoice
-                else -> Icons.Outlined.KeyboardVoice
-              },
-              stringResource(R.string.icon_descr_record_voice_message),
-              tint = when {
-                recordingTimeRange.last != 0L -> Color.White
-                stopRecOnNextClick -> MaterialTheme.colors.primary
-                allowedVoiceByPrefs -> MaterialTheme.colors.primary
-                else -> HighOrLowlight
-              },
-              modifier = Modifier
-                .size(36.dp)
-                .padding(4.dp)
-                .then(sendButtonModifier)
-            )
-          }
-          DisposableEffect(Unit) {
-            onDispose {
-              rec.stop()
-            }
-          }
+  Box(Modifier.padding(vertical = 8.dp)) {
+    val cs = composeState.value
+    val attachEnabled = !composeState.value.editing
+    val showProgress = cs.inProgress && (cs.preview is ComposePreview.ImagePreview || cs.preview is ComposePreview.VoicePreview || cs.preview is ComposePreview.FilePreview)
```

Comment:
which progress is that? this seems incorrect. isn't progress always shown on send, why is it related to what is sent?

Reply:
Who knows. Probably you wanted to show it only for "hard" operations, not the text ones, which is understandable.
This is where it's started:
https://github.com/simplex-chat/simplex-chat/commit/884231369f7c7ab3c184ccaffb994759e4ee0725#diff-3db6bd8f1ea1cbfed40e37cc5d395032d65655e6cb7440d58098556dea43a962R69

Reply:
I can make it showing like this `cs.inProgress && cs.preview !is ComposePreview.NoPreview`

Reply:
Or just `cs.inProgress && (cs.preview is ComposePreview.ImagePreview || cs.preview is ComposePreview.FilePreview)`

Reply:
It's quite annoying to see the spinner after every text message sending (image and file are ok)